rails/app/myProducts/app/controllers/products_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: %i[show update destroy]
+  # before_action :require_authorization!, only: [:show, :update, :destroy]
+
+  def index
+    products = Product.all
+    render json: { products: products }
+  end
+
+  def show
+    @product = Product.find_by(id: params[:id])
+    render json: { product: @product }
+  end
```

Comment:
É retornar 404 se o produto não for encontrado

Reply:
corrigido